taca_ngi_pipeline/deliver/deliver_mosler.py

```diff
@@ -0,0 +1,292 @@
+                raise
+
+            #memorize all samples that needs to be delivered
+            samples_to_deliver = [sentry['sampleid'] for sentry in db.project_sample_entries(
+                    db.dbcon(), self.projectid).get('samples', [])]
+            
+            # run multiple threads and store return functions
+            # http://stackoverflow.com/questions/6893968/how-to-get-the-return-value-from-a-thread-in-python
+            threads = [None] * len(samples_to_deliver)
+            results = [None] * len(samples_to_deliver)
+            thread  = 0
+            while len(samples_to_deliver) > 0:
+                # check how many samples there are in mosler sftp server
+                # create an sftp client for each sample (only one put can be done in one client. This was needed for the threaded version)
+                sftp_client = transport.open_sftp_client()
+                # move to the delivery directory in the sftp
+                sftp_client.chdir(self.expand_path(self.moslerdeliverypath))
+                # Count number of files
+                num_tar_files_mosler = len(sftp_client.listdir('.'))
+                # if the the number of remote tar files is higher the the maximum number of deliveries wait
+                if num_tar_files_mosler >= self.moslersftpmaxfiles:
+                    logger.info("More than {} files in Mosler sftp server".format(self.moslersftpmaxfiles))
+                    print "more than {} files in Mosler".format(self.moslersftpmaxfiles)
+                    sftp_client.close()
+                    # wait 10 minutes
+                    time.sleep(600)
+                    continue
+                # otherwise take next sample
+                sampleid = samples_to_deliver.pop()
+                sampleDelivererObj = MoslerSampleDeliverer(self.projectid, sampleid, sftp_client)
+                st = sampleDelivererObj.deliver_sample()
+                # initiate the thread and give it the return index
+                #threads[thread] = threading.Thread(target=sampleDelivererObj.deliver_sample_thread, args=(None, results, thread))
+                #threads[thread].start()
+                #take a nap not need to rush
+                status = (status and st) # I need to wait for the last job and then cehck the status....
+            try:
+                transport.close()
+            except Exception as e:
+                print 'Caught exception: {}: {}'.format(e.__class__, e)
+                raise
+            # query the database whether all samples in the project have been sucessfully delivered
+            if self.all_samples_delivered():
+                # this is the only delivery status we want to set on the project level, in order to avoid concurrently
+                # running deliveries messing with each other's status updates
+                self.update_delivery_status(status="DELIVERED")
+                #self.acknowledge_delivery()
```

Comment:
wouldn't it be good to use this? it is picked up by the aggregated report and included as a column